src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -0,0 +1,166 @@
+      this.direction = 'vertical';
+    }
+    if (this.focusOnLoad && isPlatformBrowser(this.platformId)) {
+      this.getCurrent().focus();
+      this._focusChange.next();
+    }
+  }
+
+  ngAfterContentChecked() {
+    this.getChildren().forEach(button => {
+      this.setTabIndex(button, -1);
+    });
+    this.setTabIndex(this.getCurrent(), 0);
+  }
+
+  @HostListener('keydown', ['$event'])
+  handleKeyboardEvent(event: any) {
+    if (this.prevClicked(event) && this._current - 1 >= 0) {
+      this.keyAction(() => {
+        this._current--;
+      });
+    } else if (this.nextClicked(event) && this._current + 1 < this.getChildren().length) {
+      this.keyAction(() => {
+        this._current++;
+      });
+    } else if (event.key === 'Home') {
+      this.keyAction(() => {
+        this._current = 0;
+      });
+    } else if (event.key === 'End') {
+      this.keyAction(() => {
+        this._current = this.getChildren().length - 1;
+      });
+    }
+  }
+
+  @HostListener('click', ['$event'])
+  handleClickEvent(event: any) {
+    const position = this.getPositionOf(event.srcElement as HTMLElement);
+    if (position > -1) {
+      this._current = position;
+    }
+  }
+
+  getPositionOf(element: HTMLElement) {
+    const children = Array.from(this.getChildren());
+    return children.indexOf(element);
+  }
+
+  moveTo(position: number) {
+    if (
+      isPlatformBrowser(this.platformId) &&
+      position >= 0 &&
+      position < this.getChildren().length &&
+      position !== this._current
+    ) {
+      this._current = position;
+      this.getCurrent().focus();
+      this._focusChange.next();
+    }
+  }
+
+  private keyAction(action: Function) {
+    if (!isPlatformBrowser(this.platformId)) {
+      return;
+    }
+    this.setTabIndex(this.getCurrent(), -1);
+    action.call(this);
+    this.setTabIndex(this.getCurrent(), 0);
+    this.getCurrent().focus();
+    this._focusChange.next();
+    // Prevent window scroll on arrow key pressed
+    event.preventDefault();
+    event.stopImmediatePropagation();
+  }
+
+  private prevClicked(event: any) {
+    switch (this.direction) {
+      case 'vertical':
+        return event.key === 'ArrowUp';
+      case 'horizontal':
+        return event.key === 'ArrowLeft';
+      case 'both':
+        return event.key === 'ArrowUp' || event.key === 'ArrowLeft';
+      default:
+        return false;
+    }
+  }
+
+  private nextClicked(event: any) {
+    switch (this.direction) {
+      case 'vertical':
+        return event.key === 'ArrowDown';
+      case 'horizontal':
+        return event.key === 'ArrowRight';
+      case 'both':
+        return event.key === 'ArrowDown' || event.key === 'ArrowRight';
+      default:
+        return false;
+    }
+  }
+
+  private getChildren(): HTMLElement[] {
+    return this.elementRef.nativeElement.querySelectorAll('button.btn') || [];
+  }
+
+  private getCurrent(): HTMLElement {
+    const children = this.getChildren();
```

Comment:
Answered above, on the 1st question. (it may be hidden if you're at the code tab, as this part of the code got changed; then you should find it in the conversation tab)

Reply:
Found it. Thanks!